staging/src/k8s.io/apimachinery/pkg/util/yaml/decoder.go

```diff
@@ -226,9 +243,16 @@ func (d *YAMLOrJSONDecoder) Decode(into interface{}) error {
 
 			start := strings.LastIndex(js[:syntax.Offset], "\n") + 1
 			line := strings.Count(js[:start], "\n")
-			return fmt.Errorf("json: line %d: %s", line, syntax.Error())
+			return JSONSyntaxError{
+				Line: line,
+				Err:  fmt.Errorf(syntax.Error()),
+			}
 		}
 	}
+
+	if err != nil && err != io.EOF && hasPrefix([]byte(err.Error()), []byte("yaml: line")) {
```

Comment:
@deads2k since the decoder defined in [pkg/util/yaml/decoder.go](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/yaml/decoder.go#L65) is what returns the error parsed here, maybe we could return the error already typed from this method [here](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/yaml/decoder.go#L68) and [here](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/yaml/decoder.go#L74). wdyt?